src/purview.ts

```diff
@@ -43,12 +43,25 @@ import {
   generateRule,
   getAtomicProperties,
 } from "./css"
+import { JSXInternal } from "./types/jsx"
 
+const InternalComponent = Component
+
+declare module "http" {
+  interface IncomingMessage {
+    purviewState?: {
+      wsState: WebSocketState
+      idStateTrees: IDStateTree[]
+      roots?: ConnectedRoot[]
+    }
+    purviewCSSState?: CSSState
+    purviewCSSRendered?: boolean
+  }
+}
 export interface RenderOptions {
   onError?: ErrorHandler
 }
-
-export interface WebSocketOptions {
+interface WebSocketOptions {
```

Comment:
I discovered incidentally that this interface was not imported anywhere. I can revert this change if desired.